hail/src/main/scala/is/hail/variant/ReferenceGenome.scala

```diff
@@ -698,20 +705,27 @@ object ReferenceGenome {
     references(name).removeLiftover(destRGName)
   }
 
-  def importReferences(fs: FS, path: String) {
+  def readReferences(fs: FS, path: String): Array[ReferenceGenome] = {
     if (fs.exists(path)) {
       val refs = fs.listStatus(path)
+      val rgs = mutable.Set[ReferenceGenome]()
       refs.foreach { fileSystem =>
         val rgPath = fileSystem.getPath.toString
         val rg = fs.readFile(rgPath)(read)
         val name = rg.name
-        if (!ReferenceGenome.hasReference(name))
-          addReference(rg)
-        else {
-          if (ReferenceGenome.getReference(name) != rg)
-            fatal(s"'$name' already exists and is not identical to the imported reference from '$rgPath'.")
-        }
+        if (ReferenceGenome.hasReference(name) && ReferenceGenome.getReference(name) != rg)
+          fatal(s"'$name' already exists and is not identical to the imported reference from '$rgPath'.")
+        if (!rgs.contains(rg) && !hailReferences.contains(name))
+          rgs += rg
       }
+      rgs.toArray
+    } else Array()
+  }
+
+  def importReferences(fs: FS, path: String) {
```

Comment:
Where is this used?

Reply:
This is used on MatrixTable/Table read to read in any custom reference genomes that we write out with the dataset.